src/libclang/preprocessor.cpp

```diff
@@ -1024,7 +1024,8 @@ ts::optional<linemarker> parse_linemarker(position& p)
 
     std::string file_name;
     for (; !starts_with(p, "\""); p.skip())
-        file_name += *p.ptr();
+        if (*p.ptr() != '\\' || file_name.empty() || file_name.back() != '\\')
```

Comment:
Please add a comment to explain what exactly you're doing here. I don't follow the logic.

Reply:
I should change, it will much understandable to revert back using De Morgan's laws.